src/chapters/control-flow-and-collections/dictionary.rst

```diff
@@ -0,0 +1,267 @@
+to be different types for a given dictionary.
+
+We suggest you run another version of the gradebook program called ``DictionaryGradebook`` in Visual Studio Code.
+This program lives in the `csharp-web-dev-lsn2controlflowandcollections <https://github.com/LaunchCodeEducation/csharp-web-dev-lsn2controlflowandcollections>`_ repository.
+If you haven't forked and cloned the repository, you should do so now.
+
+.. sourcecode:: csharp
+   :lineno-start: 10
+
+   Dictionary<string, double> students = new Dictionary<string, double>();
+   string newStudent;
+
+   Console.WriteLine("Enter your students (or ENTER to finish):");
+
+   // Get student names and grades
+   do
+   {
+      Console.WriteLine("Student: ");
+      string input = Console.ReadLine();
+      newStudent = input;
+
+      if (!Equals(newStudent, "")) {
+         Console.WriteLine("Grade: ");
+         input = Console.ReadLine();
+         double newGrade = double.Parse(input);
+         students.Add(newStudent, newGrade);
+
+         // Read in the newline before looping back
+         Console.ReadLine();
+      }
+
+   } while(!Equals(newStudent, ""));
+
+   // Print class roster
+   Console.WriteLine("\nClass roster:");
+   double sum = 0.0;
+
+   foreach (KeyValuePair<string, double> student in students) {
+      Console.WriteLine(student.Key + " (" + student.Value + ")");
+      sum += student.Value;
+   }
+
+   double avg = sum / students.Count;
+   Console.WriteLine("Average grade: " + avg);
+
+
+Notice how a ``Dictionary`` called ``students`` is declared on line 11:
+
+.. sourcecode:: C#
+   :lineno-start: 11
+
+   Dictionary<string, double> students = new Dictionary<string, double>();
+
+Here, ``<string, double>`` defines the data types for this dictionary's
+``<key, value>`` pairs.
+
+We can add a new item with a ``.Add()`` method, specifying both key and
+value:
+
+.. sourcecode:: csharp
+   :lineno-start: 26
+
+   students.Add(newStudent, newGrade);
+
+And while we don’t do so in this example, we may also access ``Dictionary``
+elements using bracket notation. If we had a key/value pair of
+``"jesse"/4.0`` in the ``students`` dictionary, we could access the grade with:
+
+.. sourcecode:: csharp
+
+   double jesseGrade = students["jesse"];
+
+Variables may be used to access elements:
+
+.. sourcecode:: csharp
+   :linenos:
+
+   string name = "jesse";
+   double jesseGrade = students[name];
+
+Looping through a dictionary is slightly more complex than it is for ordered lists.
+Let’s look at the ``foreach`` loop from this example:
+
+.. sourcecode:: csharp
+   :lineno-start: 38
+
+   for (KeyValuePair<string, double> student in students) {
+      Console.WriteLine(student.Key + " (" + student.Value + ")");
+      sum += student.Value;
+   }
+
+.. index:: ! KeyValuePair<T,T>
+
+The iterator variable, ``student``, is of type
+``KeyValuePair<string, double>``. The class ``KeyValuePair`` is specifically
```

Comment:
if we add this to the index, we should give it the **bold** treatment, no?

Reply:
Done!